utilities/new_client.py

```diff
@@ -0,0 +1,297 @@
+    directory = os.path.dirname(path)
+    if not os.path.isdir(directory):
+        os.makedirs(directory)
+    synth.dump(path)
+
+def write_pom(template: str, path: str, ctx: Context, version: str) -> None:
+    template = ctx.jinja_env.get_template(template)
+    pom = template.stream(
+        api_version=ctx.api_version,
+        description=ctx.description,
+        name=ctx.name,
+        service=ctx.service,
+        version=version
+    )
+    directory = os.path.dirname(path)
+    if not os.path.isdir(directory):
+        os.makedirs(directory)
+    pom.dump(path)
+
+def run_synthtool(ctx: Context) -> None:
+    subprocess.run(
+        [sys.executable, 'synth.py'],
+        check=True,
+        cwd=ctx.path(f'google-cloud-clients/{ctx.google_cloud_artifact}')
+    )
+
+def update_stub_packages(ctx: Context) -> None:
+    # open google-cloud-clients/pom.xml and fix the Stub packages list
+    pom = ctx.path('google-cloud-clients/pom.xml')
+    tree = etree.parse(pom)
+
+    grpc_artifacts = [v.module for v in ctx.versions if v.module.startswith('grpc-')]
+    stub_classes = []
+    for artifact in grpc_artifacts:
+        m = re.match('grpc-google-cloud-(.*)-(v.*)', artifact)
+        stub_classes.append(f'com.google.cloud.{m[1]}.{m[2]}.stub')
+
+    for group in tree.findall('.//{http://maven.apache.org/POM/4.0.0}group'):
+        if group.find('{http://maven.apache.org/POM/4.0.0}title').text == 'Stub packages':
+            group.find('{http://maven.apache.org/POM/4.0.0}packages').text = ':'.join(stub_classes)
+
+    tree.write(pom, pretty_print=True, xml_declaration=True, encoding='utf-8')
+
+def write_readme(ctx: Context) -> None:
+    template = ctx.jinja_env.get_template('README.md')
+    pom = template.stream(
+        api_version=ctx.api_version,
+        description=ctx.description,
+        name=ctx.name,
+        service=ctx.service,
+        version=ctx.google_cloud_version
+    )
+    path = ctx.path(f'google-cloud-clients/{ctx.google_cloud_artifact}/README.md')
+    directory = os.path.dirname(path)
+    if not os.path.isdir(directory):
+        os.makedirs(directory)
+    pom.dump(path)
+
+
+parser = argparse.ArgumentParser(description='Create a new client')
+parser.add_argument('-v', required=True, help='API version (i.e. v1)')
+parser.add_argument('-c', required=True, help='Path to config in googleapis/googleapis')
+parser.add_argument('-s', required=True, help='Service name')
+args = parser.parse_args()
+
+ctx = Context(
+    api_version=args.v,
+    service=args.s,
+    artman_config=args.c
+)
+
+add_to_versions(ctx)
+write_synthfile(ctx)
+run_synthtool(ctx)
+write_pom(
+    ctx=ctx,
+    template="cloud_pom.xml",
+    path=ctx.path(f'google-cloud-clients/{ctx.google_cloud_artifact}/pom.xml'),
+    version=ctx.google_cloud_version.current
+)
+add_module_to_pom(
+    pom=ctx.path('google-cloud-clients/pom.xml'),
+    module_name='google-cloud-iamcredentials'
+)
+add_dependency_management_to_pom(
```

Comment:
Could all this be done with a list and a loop instead?

Reply:
I could put each combination of dependency name, version, and target pom into a list, but I don't think it makes it less readable.